apps/mobile/src/components/private-text.tsx

```diff
@@ -0,0 +1,36 @@
+import { Pressable } from 'react-native';
+
+import { useSettings } from '@/store/settings/settings';
+import { usePrivacyMode } from '@/store/settings/settings.read';
+
+import { Text, TextProps } from '@leather.io/ui/native';
+
+const defaultPrivateText = '***';
+
+interface PrivateTextProps {
+  color?: TextProps['color'];
+  privateText?: string;
+  text: string;
+  variant?: TextProps['variant'];
+}
+export function PrivateText({
+  color,
+  privateText = defaultPrivateText,
+  text,
+  variant,
+}: PrivateTextProps) {
+  const { changePrivacyModePreference, privacyModePreference } = useSettings();
+  const isPrivate = usePrivacyMode();
+
+  function onTogglePrivacyMode() {
+    changePrivacyModePreference(privacyModePreference === 'visible' ? 'hidden' : 'visible');
+  }
+
+  return (
+    <Pressable onPress={() => onTogglePrivacyMode()}>
```

Comment:
*thought*: Another good candidate for `useCallback`?

Reply:
See comment above, I'm not sure `useCallback` is nec here?